### PR TITLE
Revise scheduling-related docs

### PR DESCRIPTION
**Pull request by sftim**
- [Move](https://github.com/kubernetes/website/commit/42061a2c7313995786e43774c07ef2722f03f55c) Pod Topology Spread Constraints into scheduling section
- [Reword](https://github.com/kubernetes/website/commit/3736c29c47a3f18a7c391cecbc8a3da9b5393ad2)  topic Assigning Pods to Nodes
- [Improve](https://github.com/kubernetes/website/commit/407ae4dd36ba6fffed3688cc611e0b3656309e80)  Pod Topology Spread Constraints concept

and some related changes. There's an online [preview](https://deploy-preview-33792--kubernetes-io-main-staging.netlify.app/docs/concepts/scheduling-eviction/).

/sig scheduling 
/label refactor
/language en

### CODE REVIEW

**Comment by sftim** [comment]
May conflict with changes in https://github.com/kubernetes/website/pull/34072

**Comment by kbhawkey** [comment]
Ping @kubernetes/sig-scheduling-misc

**Comment by sftim** [comment]
BTW, if folks want to commit to keeping https://kubernetes.io/blog/2020/05/introducing-podtopologyspread/ up to date, SIG Docs can help make that happen and can then remove the warning about stale content from that article.

File: content/en/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 1]
```diff
@@ -11,24 +11,27 @@ weight: 20
...
 There are several ways to do this and the recommended approaches all use
 [label selectors](/docs/concepts/overview/working-with-objects/labels/) to facilitate the selection.
-Generally such constraints are unnecessary, as the scheduler will automatically do a reasonable placement
+Often, you do not need to set any such constraints; the
+{{< glossary_tooltip text="scheduler" term_id="kube-scheduler" >}}  will automatically do a reasonable placement
 (for example, spreading your Pods across nodes so as not place Pods on a node with insufficient free resources).
 However, there are some circumstances where you may want to control which node
-the Pod deploys to, for example, to ensure that a Pod ends up on a node with an SSD attached to it, or to co-locate Pods from two different
-services that communicate a lot into the same availability zone.
+the Pod deploys to, for example, to ensure that a Pod ends up on a node with an SSD attached to it,
```
Comment (kbhawkey) [inline content/en/docs/concepts/scheduling-eviction/assign-pod-node.md:23]: nit: You could break up this sentence

File: content/en/docs/concepts/scheduling-eviction/assign-pod-node.md [hunk 2]
```diff
@@ -430,6 +435,10 @@ where each web server is co-located with a cache, on three separate nodes.
 | *webserver-1*        |   *webserver-2*     |    *webserver-3*   |
 |  *cache-1*           |     *cache-2*       |     *cache-3*      |
 
+The overall effect is that each cache instance is likely to be accessed by a single client, that
+is running on the same node. This approach aims to minimise both skew (imbalanced load) and latency.
```
Comment (kbhawkey) [inline content/en/docs/concepts/scheduling-eviction/assign-pod-node.md:439]: nit: minimize

**Comment by ahg-g** [comment]
/assign 

I will take a look from the scheduling sig

**Review by ahg-g — COMMENTED** [review]
Thanks!

File: content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md [hunk 3]
```diff
@@ -0,0 +1,570 @@
+---
+title: Pod Topology Spread Constraints
```
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md:2]: I moved this _and_ tweaked it. GitHub doesn't know how to represent that well. The easiest way to review is to look at the changes in https://github.com/kubernetes/website/pull/33792/commits/97f80110fdc6e57f14dedf230c1854a4aba74439 (wrapping) and https://github.com/kubernetes/website/pull/33792/commits/9fe1f68718e8d7fbdb5c40c024d05210483227d9 (text changes, including some moves within the page).

[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/concepts/scheduling-eviction/topology-spread-constraints.md:2]: You can also compare the [original](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/) to the [preview](https://deploy-preview-33792--kubernetes-io-main-staging.netlify.app/docs/concepts/scheduling-eviction/topology-spread-constraints/).

**Comment by windsonsea** [comment]
Looks good to me. Let's kick it in while two months are passed.
/lgtm

**Pull request merged**